c/xen_common.c

```diff
@@ -885,22 +885,27 @@ static int count_children(xmlNode *n, const char *name)
 static void destring(xen_session *s, xmlChar *name, const abstract_type *type,
                      void *value)
 {
-    switch (type->typename)
+    if (type->typename == STRING)
     {
-    case STRING:
-        *((char **)value) = xen_strdup_((const char *)name);
-        break;
-
-    case INT:
-        *((int64_t *)value) = atoll((const char *)name);
-        break;
-
-    case FLOAT:
-        *((double *)value) = atof((const char *)name);
-        break;
-
-    default:
-        server_error(s, "Invalid Map key type");
+        *((char **) value) = xen_strdup_((const char *) name);
+    }
+    else if (type->typename == INT)
+    {
+        *((int64_t *) value) = atoll((const char *) name);
+    }
+    else if (type->typename == FLOAT)
+    {
+        *((double *) value) = atof((const char *) name);
+    }
+    else if (type->typename == ENUM)
+    {
+       *((int *)value) = type->enum_demarshaller(s, (const char *)name);
+    }
+    else
+    {
+        char buf[256];
+        snprintf(buf, sizeof (buf), "%s", name);
```

Comment:
I think it would be great, if we could change this to "Invalid Map key type: %s".
I wouldn't have had a chance to find this bit of code without the Invalid Map key type hint.